### ISSUE TITLE
pointsToStroke: Return a shape rather than a geometry

### ISSUE THREAD

**Issue by arcanis** [mrdoob/three.js#21326]
**Is your feature request related to a problem? Please describe.**

I'd like to create a map from an svg image. For this to work, I'd like to generate outlines for the map, and use the ExtrudeGeometry in order to create the walls.

**Describe the solution you'd like**

The `pointsToStroke` function seems to be what I'm looking for, however it's currently hardcoded to only return BufferGeometry. Given that the rest of the SVG loader works with shapes, I feel like it'd make sense API-wise that `pointsToStroke` would do so as well.

**Describe alternatives you've considered**

Tried a lot of alternatives to generate the outline, but the overhead w/ Three.js is significant. Since there are primitives supposed to do just that already, I'd prefer to only have to use them.

**Comment by yomboprime** [comment]
Unfortunately the ```pointsToStroke``` algorithm doesn't generate always a beautiful mesh that can be extruded: sometimes there are overlapping triangles, as shown in this zoomed in image of a hair of the Tiger (strokes in wireframe mode):
![imagen](https://user-images.githubusercontent.com/2453652/108628135-6c21d500-7459-11eb-8cb5-bc35080eae17.png)

Nevertheless, there is a solution: You can generate a fill shape (not stroke) with the form of the desired cross section of the walls, and use the shapes to extrude rather than converting directly to plane mesh.

Inkscape, for example, can convert a path stroke to a fill shape (Menu Path --> Option Border to Path)

**Comment by arcanis** [comment]
> Nevertheless, there is a solution: You can generate a fill shape (not stroke) with the form of the desired cross section of the walls, and use the shapes to extrude rather than converting directly to plane mesh.

Indeed, I could generate the wall map svg rather than the one with the floor map. I was however hoping I could find a way to do with only the floor map (and derive the wall map from it), as it would be far easier to edit for the user 🤔

I'm experimenting with the following code, which almost work:

```ts
const group = new THREE.Group();

for (const path of floorShape.paths) {
  for (const subpath of path.subPaths) {
    const output: number[] = [];
    SVGLoader.pointsToStrokeWithBuffers(subpath.getPoints(), strokeStyle, undefined, undefined, output, [], []);

    const shape = new THREE.Shape();
    for (let t = 0, T = output.length / 3; t < T; ++t) {
      shape.moveTo(output[t * 3 + 0], output[t * 3 + 1]);
      shape.lineTo(output[t * 3 + 3], output[t * 3 + 4]);
      shape.lineTo(output[t * 3 + 6], output[t * 3 + 7]);
    }

    const geometry = geometryFactory(shape);
    const mesh = new THREE.Mesh(geometry, material);
    group.add(mesh);
    return group;
  }
}
```

Unfortunately, it seems to generate various weird moveTo instructions, I'm not sure why... For example, in the following picture, the screw-like form is supposed to be a room, but my code causes it to be traversed by many edges.

![image](https://user-images.githubusercontent.com/1037931/108631736-98dee800-746b-11eb-9420-9c3f14531c6f.png)

**Comment by arcanis** [comment]
I tried the Inkscape approach, but unfortunately it seems the generated svg hits the problem in #16950 😔

**Comment by yomboprime** [comment]
I've got an idea. Just generate the strokes mesh with UVs coordinates, and take all the stroke points which have U == 0.0 (U == 1.0 is the other side, U = 0.5 is for points in the inside, and V varies from 0.0 to 1.0 along the path)

With those points you should be able to construct an extrudable outline.

**Comment by LittleSound** [comment]
I am also looking forward to this feature. Extrusion outlines are very useful. I am writing a tool for converting SVG to 3D models, so I can't control what content the user's SVG includes.
